### ISSUE TITLE
New Youtube Format Bug

### ISSUE THREAD

**Issue by n0753w** [Anarios/return-youtube-dislike#1037]
### Have you tried to find similar issues (open or recently closed)?

- [X] Yes, this issue is not a duplicate

### Browser

MS Edge

### Browser Version

Version 121.0.2277.128 

### Extension or Userscript?

Extension

### Extension/Userscript Version

v3.0.0.14

### Video link where you see the problem

Any Youtube video

### What happened?

The dislike counter doesn't show with the new ugly UI. Is it possible to make a workaround or does this addon only work with the "normal" UI?
![image](https://github.com/Anarios/return-youtube-dislike/assets/138560355/6992c503-8774-4dd5-9d23-f1bca81ffa82)

### How to reproduce/recreate?

This only happens because of the new UI. If someone else has said UI, then most addons will fail to work. 

### Will you be available for follow-up questions to help developers diagnose & fix the issue?

Yes

**Comment by FilipLac** [comment]
I also have the same issue.

**Comment by cocoKoniQ** [comment]
Can confirm, I have the same issue.

**Comment by CobaltM** [comment]
yep same issue

**Comment by greatwolf** [comment]
Same issue here.

**Comment by CHlLL7** [comment]
Had the same issue with the new UI but as a workaround: Installing "Enhancer for Youtube" brings back the old UI and Return Youtube dislikes will work again

**Comment by n0753w** [comment]
> CHlLL7

Yeah I also have that extension. It's great. About a week or so ago, the guy who made the YT Enhancer made an update to prevent the ugly ui from occurring, which is great.

**Comment by Couchy** [comment]
Is this project dead? I could fix the UI, but there's not much point if the backend is also unmaintained.

**Comment by n0753w** [comment]
Everything is back to normal for now on my side at least. YT likes to dish out experimental formats randomly.

**Comment by RocketRobz** [comment]
The new format has rolled out recently on my side, and using "Enhancer for Youtube" also restores the dislikes on my side.

**Comment by theblackhole** [comment]
This issue has been fixed with the latest release ([v3.0.0.16](https://github.com/Anarios/return-youtube-dislike/releases/tag/v3.0.0.16)) which is already available for Edge/Chrome in the Chrome Webstore (Firefox soon)

With this new release, there's also a minor issue where the buttons are aligned to the right when the extension is enabled. I provided a fix (#1047) and it should be available in a future update.

**Comment by Orevin** [comment]
It looks like this issue resurfaced (extension 3.0.0.16)

**Comment by archon810** [comment]
Yeah, 3.0.0.16 is indeed not working.

**Comment by leakedmemory** [comment]
Currently I'm on version 3.0.0.14, on LibreWolf 127.0-1, and have the same issue.

**Comment by leakedmemory** [comment]
I'm trying to solve this issue, but I don't really know where to look. What I found is that the div that contains the number of dislikes it not being rendered in normal screen size, but if you decrease the size of the screen it'll be rendered correctly. Also, if I understood it correctly, the plugin is not even doing anything on normal size, because there's no logs.

Works fine when screen size is reduced:
![small](https://github.com/Anarios/return-youtube-dislike/assets/76412614/1de54c95-6c94-43e9-8bfd-af3d157367cb)

It's bugged with normal screen size:
![normal](https://github.com/Anarios/return-youtube-dislike/assets/76412614/0690f1d8-1370-4582-86d9-21ac54e5292c)

**Comment by leakedmemory** [comment]
This problem is related to the line 22 of the `Extensions/combined/ryd.content-script.js` file. With the new layout change, when the screen is in normal size, there's no valid `offsetParent`, which is evaluated into `null` and the code never gets inside the if statement.

```javascript
if (isShorts() || (getButtons()?.offsetParent && isVideoLoaded())) {
```

This happens because before the change, it would evaluate to a div with id `#below`, which still exists, but in the primary column. The primary column is the column with the video player and the recommendations below it, and the secondary column is the one with the title, buttons, comments and so on. When the size is shrunk, the secondary column's content goes to the primary column, finally evaluating the `getButtons()?.offsetParen` to true and running the code inside the statement.

**Comment by leakedmemory** [comment]
Also when you put the window back in normal size, it happens the bug commented in PR #1047.

**Comment by valch85** [comment]
yep, the extension version 3.0.0.16 for the FF doesn't show a dislike count =(
![CleanShot 2024-07-03 at 13 07 41@2x](https://github.com/Anarios/return-youtube-dislike/assets/1758849/78b2fb94-fc72-4ac3-ab68-f31603f1b599)

**Comment by Anarios** [comment]
Fixed in [v4.0.5](https://github.com/Anarios/return-youtube-dislike/releases/tag/v4.0.5). This includes the layout-detection correction discussed above: rendered reaction controls no longer need an offset parent for initialization, so the dislike count can appear in the wider layout as well as the narrow layout. Action-row alignment and overflow handling have also been updated.

Please update the extension and reload YouTube. For Firefox users, the store update is still awaiting Mozilla review.
